src/App.js

```diff
@@ -1,50 +1,56 @@
-import './App.css';
+import './style/App.css';
 import { useEffect, useState } from "react";
-import { ShowPokemon } from './showPokemon';
+import { ShowPokemon } from './components//showPokemon';
+import { Button } from './components/buttons';
+import Loading from 'react-simple-loading';
 
 function App() {
 
 const [currentPageUrl, setCurrentPageUrl] =useState("https://pokeapi.co/api/v2/pokemon")
 const [pokeName, setPokeName] = useState([])
-const [allPokemon, setAllPokemon] = useState([])
 const [nextPage, setNextPage] = useState('')
+const [prevPage, setPrevPage] = useState('')
+const [loading, setLoading] =useState(true)
 
 
 const getPokemons = async () => {
-  let res = await fetch(currentPageUrl)
-  let data = await res.json()
-  setNextPage(data.next) // display 20 more pokemon (btn in future)
-  setPokeName(data.results.map(p => p.name))
-
-  function createPokemonObject(result) {
-    result.forEach(async (pokemon) => {
-      let res = await fetch(`https://pokeapi.co/api/v2/pokemon/${pokemon}`)
-      let data = await res.json()
-      setAllPokemon(currentList => [...currentList, data])
-    })
+  try{
+    setLoading(true)
+    let res = await fetch(currentPageUrl)
+    let data = await res.json()
+    setNextPage(data.next)
+    setPrevPage(data.previous)
+    setPokeName(data.results.map(p => p.name))
+    setLoading(false)
+  } catch (error) {
+      alert("Błąd ładowania danych", error)
   }
-
-  createPokemonObject(pokeName)
+  
 }
 
-
 useEffect(() => {
   getPokemons()
-}, [])
-
+}, [currentPageUrl])
 
-
-console.log(allPokemon) 
-console.log(pokeName) // display on first load
+if (loading) return (
+  <div>
+    <Loading />
+  </div>
+)
 
   return (
     <>
       <div className="main">
         <div className="pokedex">
           <h1 className='title'>POKEDEX</h1>
+          <div className="search">
+            <input 
+            type="search" />
+          </div>
+          <Button prevPage={prevPage} nextPage={nextPage} setCurrentPageUrl={setCurrentPageUrl}/>
           <div className="poke-list">
-            {allPokemon.map(data =>
-              <ShowPokemon data={data}/>
+            {pokeName.map(name =>
+              <ShowPokemon Pokemon={name}/>
```

Comment:
Props's name starts with small letter

```suggestion
              <ShowPokemon pokemon={name}/>
```